Concuerror 0.17 (a62e2f6) started at 09 Aug 2017 12:46:40
 Options:
  [{after_timeout,infinity},
   {assertions_only,false},
   {assume_racing,false},
   {depth_bound,500},
   {disable_sleep_sets,false},
   {dpor,optimal},
   {entry_point,{after_test_reduced,after_test_reduced,[]}},
   {files,["/home/stavros/git/Concuerror/tests/suites/dpor_tests/src/after_test_reduced.erl"]},
   {ignore_error,[]},
   {instant_delivery,true},
   {interleaving_bound,infinity},
   {keep_going,true},
   {non_racing_system,[]},
   {print_depth,20},
   {quiet,true},
   {scheduling,round_robin},
   {scheduling_bound_type,none},
   {show_races,false},
   {strict_scheduling,false},
   {symbolic_names,true},
   {timeout,infinity},
   {treat_as_normal,[]},
   {use_receive_patterns,false}]
################################################################################
Interleaving #1
--------------------------------------------------------------------------------
Errors found:
* Blocked at a 'receive' ("deadlocked"; other processes have exited):
    P in after_test_reduced.erl line 13
--------------------------------------------------------------------------------
Event trace:
   1: P: P.1 = erlang:spawn(erlang, apply, [#Fun<after_test_reduced.'-after_test_reduced/0-fun-0-'.0>,[]])
    in erlang.erl line 2673
   2: P: P.2 = erlang:spawn(erlang, apply, [#Fun<after_test_reduced.'-after_test_reduced/0-fun-1-'.0>,[]])
    in erlang.erl line 2673
   3: P: receive timeout expired after 0 ms
    in after_test_reduced.erl line 23
   4: P.1: ok2 = P ! ok2
    in after_test_reduced.erl line 10
   5: P.1: exits normally
   6: P.2: ok2 = P ! ok2
    in after_test_reduced.erl line 11
   7: P.2: exits normally
   8: P: receives message (ok2)
    in after_test_reduced.erl line 19
   9: P: receives message (ok2)
    in after_test_reduced.erl line 19
  10: P: receive timeout expired after 0 ms
    in after_test_reduced.erl line 30
################################################################################
Interleaving #2
--------------------------------------------------------------------------------
Errors found:
* Blocked at a 'receive' ("deadlocked"; other processes have exited):
    P in after_test_reduced.erl line 13
--------------------------------------------------------------------------------
Event trace:
   1: P: P.1 = erlang:spawn(erlang, apply, [#Fun<after_test_reduced.'-after_test_reduced/0-fun-0-'.0>,[]])
    in erlang.erl line 2673
   2: P: P.2 = erlang:spawn(erlang, apply, [#Fun<after_test_reduced.'-after_test_reduced/0-fun-1-'.0>,[]])
    in erlang.erl line 2673
   3: P: receive timeout expired after 0 ms
    in after_test_reduced.erl line 23
   4: P.2: ok2 = P ! ok2
    in after_test_reduced.erl line 11
   5: P.2: exits normally
   6: P: receives message (ok2)
    in after_test_reduced.erl line 19
   7: P.1: ok2 = P ! ok2
    in after_test_reduced.erl line 10
   8: P.1: exits normally
   9: P: receives message (ok2)
    in after_test_reduced.erl line 19
  10: P: receive timeout expired after 0 ms
    in after_test_reduced.erl line 30
################################################################################
Interleaving #3
--------------------------------------------------------------------------------
Errors found:
* Blocked at a 'receive' ("deadlocked"; other processes have exited):
    P in after_test_reduced.erl line 13
--------------------------------------------------------------------------------
Event trace:
   1: P: P.1 = erlang:spawn(erlang, apply, [#Fun<after_test_reduced.'-after_test_reduced/0-fun-0-'.0>,[]])
    in erlang.erl line 2673
   2: P: P.2 = erlang:spawn(erlang, apply, [#Fun<after_test_reduced.'-after_test_reduced/0-fun-1-'.0>,[]])
    in erlang.erl line 2673
   3: P.1: ok2 = P ! ok2
    in after_test_reduced.erl line 10
   4: P.1: exits normally
   5: P.2: ok2 = P ! ok2
    in after_test_reduced.erl line 11
   6: P.2: exits normally
   7: P: receives message (ok2)
    in after_test_reduced.erl line 23
   8: P: receives message (ok2)
    in after_test_reduced.erl line 23
   9: P: receive timeout expired after 0 ms
    in after_test_reduced.erl line 30
################################################################################
Interleaving #4
--------------------------------------------------------------------------------
Errors found:
* Blocked at a 'receive' ("deadlocked"; other processes have exited):
    P in after_test_reduced.erl line 13
--------------------------------------------------------------------------------
Event trace:
   1: P: P.1 = erlang:spawn(erlang, apply, [#Fun<after_test_reduced.'-after_test_reduced/0-fun-0-'.0>,[]])
    in erlang.erl line 2673
   2: P: P.2 = erlang:spawn(erlang, apply, [#Fun<after_test_reduced.'-after_test_reduced/0-fun-1-'.0>,[]])
    in erlang.erl line 2673
   3: P.1: ok2 = P ! ok2
    in after_test_reduced.erl line 10
   4: P.1: exits normally
   5: P: receives message (ok2)
    in after_test_reduced.erl line 23
   6: P: receive timeout expired after 0 ms
    in after_test_reduced.erl line 23
   7: P.2: ok2 = P ! ok2
    in after_test_reduced.erl line 11
   8: P.2: exits normally
   9: P: receives message (ok2)
    in after_test_reduced.erl line 19
  10: P: receive timeout expired after 0 ms
    in after_test_reduced.erl line 30
################################################################################
Interleaving #5
--------------------------------------------------------------------------------
Errors found:
* Blocked at a 'receive' ("deadlocked"; other processes have exited):
    P in after_test_reduced.erl line 13
--------------------------------------------------------------------------------
Event trace:
   1: P: P.1 = erlang:spawn(erlang, apply, [#Fun<after_test_reduced.'-after_test_reduced/0-fun-0-'.0>,[]])
    in erlang.erl line 2673
   2: P: P.2 = erlang:spawn(erlang, apply, [#Fun<after_test_reduced.'-after_test_reduced/0-fun-1-'.0>,[]])
    in erlang.erl line 2673
   3: P.2: ok2 = P ! ok2
    in after_test_reduced.erl line 11
   4: P.2: exits normally
   5: P: receives message (ok2)
    in after_test_reduced.erl line 23
   6: P: receive timeout expired after 0 ms
    in after_test_reduced.erl line 23
   7: P.1: ok2 = P ! ok2
    in after_test_reduced.erl line 10
   8: P.1: exits normally
   9: P: receives message (ok2)
    in after_test_reduced.erl line 19
  10: P: receive timeout expired after 0 ms
    in after_test_reduced.erl line 30
################################################################################
Interleaving #6
--------------------------------------------------------------------------------
Errors found:
* Blocked at a 'receive' ("deadlocked"; other processes have exited):
    P in after_test_reduced.erl line 13
--------------------------------------------------------------------------------
Event trace:
   1: P: P.1 = erlang:spawn(erlang, apply, [#Fun<after_test_reduced.'-after_test_reduced/0-fun-0-'.0>,[]])
    in erlang.erl line 2673
   2: P: P.2 = erlang:spawn(erlang, apply, [#Fun<after_test_reduced.'-after_test_reduced/0-fun-1-'.0>,[]])
    in erlang.erl line 2673
   3: P.2: ok2 = P ! ok2
    in after_test_reduced.erl line 11
   4: P.2: exits normally
   5: P: receives message (ok2)
    in after_test_reduced.erl line 23
   6: P.1: ok2 = P ! ok2
    in after_test_reduced.erl line 10
   7: P.1: exits normally
   8: P: receives message (ok2)
    in after_test_reduced.erl line 23
   9: P: receive timeout expired after 0 ms
    in after_test_reduced.erl line 30
################################################################################
Exploration completed!
################################################################################
Tips:
--------------------------------------------------------------------------------
Increase '--print_depth' if output/graph contains "...".

################################################################################
Info:
--------------------------------------------------------------------------------
Instrumented after_test_reduced
Instrumented io_lib
Instrumented gen_server
Instrumented erlang
You can see pairs of racing instructions (in the report and '--graph') with '--show_races true'
Continuing after error (-k)

################################################################################
Done at 09 Aug 2017 12:46:40 (Exit status: error)
  Summary: 6 errors, 6/6 interleavings explored
